src/web/components/StickyBottomBanner/BrazeBanner.tsx

```diff
@@ -61,15 +63,34 @@ export const canShow = async (
         );
 
         appboy.initialize(apiKey, {
+            enableLogging: false,
             noCookies: true,
             baseUrl: 'https://sdk.fra-01.braze.eu/api/v3',
             sessionTimeoutInSeconds: 1,
             minimumIntervalBetweenTriggerActionsInSeconds: 0,
         });
 
         return new Promise((resolve) => {
-            appboy.subscribeToInAppMessage((message) => {
-                const meta = (message as any).extras;
+            appboy.subscribeToInAppMessage((message: any) => {
+                const meta = message.extras;
+
+                buttonHandler = (buttonId: number) => {
+                    const thisButton = new appboy.InAppMessageButton(
+                        `Button: ID ${buttonId}`,
+                        undefined,
+                        undefined,
+                        undefined,
+                        undefined,
+                        undefined,
+                        buttonId,
+                    );
+                    appboy.logInAppMessageButtonClick(thisButton, message);
+                };
+
+                if (appboy) {
+                    // Log the impression with Braze
+                    appboy.logInAppMessageImpression(message);
```

Comment:
It's subtle but this will log the impression when the callback is fired, not when we actually show the banner component to the user, which is what we want. I think using the approach where this is threaded through using the `meta` object is the way to go.